zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSettingManager.java

```diff
@@ -455,11 +455,9 @@ public InterpreterSetting getDefaultInterpreterSetting(String noteId) {
 
   private List<String> getNoteInterpreterSettingBinding(String noteId) {
     LinkedList<String> bindings = new LinkedList<>();
-    synchronized (interpreterSettings) {
```

Comment:
I think this synchronized block shouldn't be removed to protect `interpreterBindings`